wine/wine.go

```diff
@@ -3,60 +3,137 @@
 package wine
 
 import (
+	"errors"
 	"io"
-	"log"
+	"os"
 	"os/exec"
+	"path/filepath"
 )
 
-// The program used for Wine.
-var Wine = "wine"
+var (
+	ErrWineRootAbs  = errors.New("wineroot is not absolute")
+	ErrWineNotFound = errors.New("wine64 not found in system or wineroot")
+)
 
 // Prefix is a representation of a wineprefix, which is where
-// WINE stores its data, which is equivalent to WINE's C:\ drive.
+// WINE stores its data and is equivalent to a C:\ drive.
 type Prefix struct {
-	// Output specifies the descendant prefix commmand's
-	// Stderr and Stdout together.
-	//
-	// Wine will always write to stderr instead of stdout,
-	// Stdout is combined just to make that certain.
-	Output io.Writer
+	// Path to a wine installation.
+	Root string
+
+	// Stdout and Stderr specify the descendant Prefix wine call's
+	// standard output and error. This is mostly reserved for logging purposes.
+	// By default, they will be set to their os counterparts.
+	Stderr io.Writer
+	Stdout io.Writer
+
+	wine string
+	dir  string
+}
+
+func (p Prefix) String() string {
+	return p.dir
+}
+
+// Wine64 returns a path to the system or wineroot's 'wine64'.
+func Wine64(root string) (string, error) {
+	var bin string
 
-	dir string
+	if root != "" && !filepath.IsAbs(root) {
+		bin = filepath.Join(root, "bin")
+		return "", ErrWineRootAbs
+	}
+
+	wine, err := exec.LookPath(filepath.Join(bin, "wine64"))
```

Comment:
Must notify all distributions of this change, some have symlinked wine64 to wine.